### PR TITLE
CVSL-1380 Adding detekt to circle ci

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: .scripts/pre-commit/pre-commit [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+for hook in "$(dirname "$0")/pre-commit-scripts"/*; do
+   "$hook"
+   RESULT=$?
+
+   if [ $RESULT != 0 ]; then
```
Comment (justin-gardiner) [inline .scripts/pre-commit/pre-commit:7]: should this be -ne like the exit status below?

**Review by gurps-singh — APPROVED** [review]

File: .scripts/pre-commit/pre-commit (hunk 1 above)
Comment (justin-gardiner) [reply .scripts/pre-commit/pre-commit:7]: actually just checked and both ways seem to work

File: .scripts/pre-commit/pre-commit-scripts/detekt.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+   echo "***********************************************"
+   echo "                 detekt passed                 "
+   echo "    No issues need fixing before committing    "
+   echo "***********************************************"
+
+   exit $EXIT_CODE
+fi
+
+rm "$OUTPUT"
```
Comment (justin-gardiner) [inline .scripts/pre-commit/pre-commit-scripts/detekt.sh:34]: It doesn't look like we can ever reach the rm as both branches of if exit

Comment (justin-gardiner) [reply .scripts/pre-commit/pre-commit-scripts/detekt.sh:26]: do you still want the exit $EXIT_CODE after the rm or doesn't it matter?

Comment (justin-gardiner) [reply .scripts/pre-commit/pre-commit-scripts/detekt.sh:26]: sorry just seen it is there!

**Review by justin-gardiner — APPROVED** [review]
lgtm

**Pull request merged**